neo4j/src/main/java/org/hibernate/ogm/datastore/neo4j/remote/transaction/impl/RemoteJtaTransactionCoordinator.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.neo4j.remote.transaction.impl;
+
+import javax.transaction.Status;
+import javax.transaction.Synchronization;
+
+import org.hibernate.ogm.datastore.neo4j.remote.impl.Neo4jClient;
+import org.hibernate.ogm.datastore.neo4j.remote.impl.RemoteNeo4jDatastoreProvider;
+import org.hibernate.ogm.transaction.impl.ForwardingTransactionCoordinator;
+import org.hibernate.ogm.transaction.impl.ForwardingTransactionDriver;
+import org.hibernate.resource.transaction.TransactionCoordinator;
+import org.hibernate.resource.transaction.spi.TransactionStatus;
+
+/**
+ * @author Davide D'Alto
+ */
+public class RemoteJtaTransactionCoordinator extends ForwardingTransactionCoordinator {
```

Comment:
As far as I know `Synchronization` does not tell me when the transaction starts, it only tells me when the before and after commit event. The reason I have all this delegation business is to provide a `TransactionDriver` bound to the remote neo4j transaction.
Let me know if I'm missing something.